### DISCUSSION TITLE
Virtualhost aliases with one FQDN and multipleroots

### DISCUSSION THREAD

**Discussion by rdgacarvalho** [projectcontour/contour#4178] [Q&A] [answered]
Hi,
I'm trying to create virtual host aliases with the following concept.

> Using contour 5.5.0 on Kubernetes EKS v1.21

The manifest main.yaml contains the main route to the service in the same namespace.

```(yaml)
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: main
  namespace: lionman
spec:
  virtualhost:
    fqdn: apis.contour.glaedr.local
  routes:
  - conditions:
    - prefix: /
    services:
    - name: cms
      port: 8080
```

cms.yaml
```(yaml)
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: cms
  namespace: lionman
spec:
  includes:
  - name: main
    namespace: lionman
  routes:
  - conditions:
    - prefix: /v1/cms/media-assets-bucket/
    enableWebsockets: true
    services:
    - name: cms-api-service
      port: 8080
  - conditions:
    - prefix: /v1/cms/process-steps-one/
    enableWebsockets: true
    services:
    - name: cms-api-service
      port: 8080
```

uploader.yaml
```(yaml)
apiVersion: projectcontour.io/v1
kind: HTTPProxy
metadata:
  name: uploader
  namespace: lionman
spec:
  includes:
  - name: main
    namespace: lionman
  routes:
  - conditions:
    - prefix: /v1/cms/upload/
    enableWebsockets: true
    services:
    - name: uploader-service-service
      port: 8080
```

The following files contain different routes for different services in the same namespace. However, I'm still getting into the *orphaned* with the message below.

`this HTTPProxy is not part of a delegation chain from a root HTTPProxy`

Is it possible to achieve the desired setup with contour nowadays?

Thank you very much.

Best,

**Comment by youngnick** [comment, marked as answer]
Thanks for asking @rdgacarvalho!

The HTTPProxy with the FQDN is the one that needs to include the child HTTPProxies, not the other way around.

Currently, you have the children including the parent, which will fail, because you can't include a HTTPProxy that has an FQDN defined.
